### PR TITLE
refactor(node-core/matrics): refactor the register version metrics function

### LINKED ISSUE #9101: move metrics support to node/metrics crate
### Describe the feature

https://github.com/paradigmxyz/reth/blob/270df3eeb807e148d317d0037b2bf1499530a1d1/crates/node-core/src/metrics/mod.rs#L1-L1

minus:

https://github.com/paradigmxyz/reth/blob/270df3eeb807e148d317d0037b2bf1499530a1d1/crates/node-core/src/metrics/version_metrics.rs#L6-L18

because these currently depend on variables that can only defined where we run the build.rs script

so this would need to be passed as an input instead or set separately

https://github.com/paradigmxyz/reth/blob/270df3eeb807e148d317d0037b2bf1499530a1d1/crates/node-core/src/metrics/prometheus_exporter.rs#L154-L154

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by kien6034**
This closes #9101

### CODE REVIEW

File: crates/node-core/src/metrics/version_metrics.rs [hunk 1]
```diff
@@ -1,18 +1,40 @@
...
-    ("target_triple", env!("VERGEN_CARGO_TARGET_TRIPLE")),
-    ("build_profile", build_profile_name()),
-];
+pub struct VersionInfo {
+    version: &str,
+    build_timestamp: &str,
+    cargo_features: &str,
+    git_sha: &str,
+    target_triple: &str,
+    build_profile: &str,
```
Comment (mattsse) [inline crates/node-core/src/metrics/version_metrics.rs:12]: can we make these pub?

if we extract metrics from node-core we'd need this as input I believe so makes sense to make these pub

File: crates/node-core/src/metrics/version_metrics.rs [hunk 2]
```diff
@@ -1,18 +1,40 @@
...
 
-const LABELS: [(&str, &str); 6] = [
-    ("version", env!("CARGO_PKG_VERSION")),
-    ("build_timestamp", env!("VERGEN_BUILD_TIMESTAMP")),
-    ("cargo_features", env!("VERGEN_CARGO_FEATURES")),
-    ("git_sha", VERGEN_GIT_SHA),
-    ("target_triple", env!("VERGEN_CARGO_TARGET_TRIPLE")),
-    ("build_profile", build_profile_name()),
-];
+pub struct VersionInfo {
```
Comment (mattsse) [inline crates/node-core/src/metrics/version_metrics.rs:6]: + debug +clone

File: crates/node-core/src/metrics/version_metrics.rs [hunk 3]
```diff
@@ -1,18 +1,40 @@
...
+            target_triple: env!("VERGEN_CARGO_TARGET_TRIPLE"),
+            build_profile: build_profile_name(),
+        }
+    }
+}
 
 /// This exposes reth's version information over prometheus.
-pub fn register_version_metrics() {
-    let _gauge = gauge!("info", &LABELS);
+pub fn register_version_metrics(version_info: VersionInfo) {
```
Comment (mattsse) [inline crates/node-core/src/metrics/version_metrics.rs:29]: can we make this a function of versioninfo instead?

**Review by mattsse — CHANGES_REQUESTED** [review]
nice,

few pedantic style nits

File: crates/node-core/src/metrics/version_metrics.rs (hunk 1 above)
Comment (kien6034) [reply crates/node-core/src/metrics/version_metrics.rs]: yeah it makes sense

**Comment by kien6034** [comment]
@mattsse tks for reviewing sir. I updated the code referencing your comments

**Comment by kien6034** [comment]
lint is failing, fixing it...

**Review by mattsse — APPROVED** [review]

**Pull request merged**